ros/processor/inventory_events_consumer.py

```diff
@@ -1,98 +1,62 @@
-
-    def __next__(self):
-        msg = self.consumer.poll()
-        if msg is None:
-            raise StopIteration
-        return msg
 
     def run(self):
         """Initialize Consumer."""
-        LOG.info(f"{self.prefix} - Processor is running. Awaiting msgs.")
+        self.LOG.info(f"{self.prefix} - Processor is running. Awaiting msgs.")
         for msg in iter(self):
-            if msg.error():
-                LOG.error(f"{self.prefix} - Consumer error: {msg.error()}")
-                kafka_failures.labels(reporter=self.reporter).inc()
-                raise KafkaException(msg.error())
-
-            account = None
-            host_id = None
-            org_id = None
-            try:
-                msg = json.loads(msg.value().decode("utf-8"))
-                event_type = msg['type']
-                if event_type == 'delete':
-                    account = msg['account']
-                    host_id = msg['id']
-                    org_id = msg['org_id']
-                else:
-                    account = msg['host']['account']
-                    host_id = msg['host']['id']
-                    org_id = msg['host'].get('org_id')
-
-                if event_type in self.event_type_map.keys():
-                    handler = self.event_type_map[event_type]
-                    handler(msg)
-                else:
-                    LOG.info(
-                        f"{self.prefix} - Event Handling is not found for event {event_type}"
-                    )
-            except json.decoder.JSONDecodeError:
-                kafka_failures.labels(reporter=self.reporter).inc()
-                LOG.error(
-                    f"{self.prefix} - Unable to decode kafka message: {msg.value()}"
-                )
-            except Exception as err:
-                processor_requests_failures.labels(
-                    reporter=self.reporter, org_id=org_id
-                ).inc()
-                LOG.error(
-                    f"{self.prefix} - An error occurred during message processing: {repr(err)} "
-                    f"in the system {host_id} created from account: {account} and org_id: {org_id}"
+            self.check_msg_for_err(msg, self.prefix, self.reporter)
+            decoded_msg = self.decode_and_load_json(msg)
+            event_type, account, host_id, org_id = extract_attr(decoded_msg)
+            processing_err = f"\nin the system {host_id} created from account: {account} and org_id: {org_id}"
+            if event_type in self.event_type_map.keys():
+                handler = self.event_type_map[event_type]
+                self.try_to_handle_msg(handler(decoded_msg), org_id, processing_err)
+            else:
+                self.LOG.info(
+                    f"{self.prefix} - Event Handling is not found for event {event_type}"
                 )
-            finally:
-                self.consumer.commit()
-        LOG.warning("Stopping inventory consumer")
-        self.consumer.close()
```

Comment:
Hello @patilsuraj767  - 
In inventory related processor, we are explicitly closing the consumer whereas in engine related processor, we are not closing it. 
Could you share few inputs & any scenario you can think of might cause issue when we close on engine side too?

Reply:
closing the consumer connection is good practice but I guess not mandatory. Basically as per me closing the connection just speed up the process. Example - Consider there is a pod consuming msgs from Kafka but suddenly pod dies for some reason without closing the connection and then OpenShift spin up the new pod but new pod will not be quickly able to connect to Kafka topic till the timeout (30s default) happens and Kafka broker terminates the previous connection.

Reply:
This makes it better then! I would close the connections for both! Thanks for inputs!